File: java/app/src/main/java/com/youknow/movie/ui/MainActivity.java
package com.youknow.movie.ui;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.youknow.movie.R;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

public class MainActivity extends AppCompatActivity {

    public static final String MOVIE_ID = "movie_id";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final ViewPager mainViewPager = findViewById(R.id.mainViewPager);
        final BottomNavigationView bottomNavigation = findViewById(R.id.bottomNavigation);

        mainViewPager.setAdapter(new BottomNavigationAdapter(getSupportFragmentManager()));

        bottomNavigation.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                switch (menuItem.getItemId()) {
                    case R.id.action_now_playing:
                        mainViewPager.setCurrentItem(0);
                        return true;
                    case R.id.action_upcoming:
                        mainViewPager.setCurrentItem(1);
                        return true;
                    default:
                        return false;
                }
            }
        });

        mainViewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                bottomNavigation.getMenu().getItem(position).setChecked(true);
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
    }
}

File: kotlin-databinding-coroutine-aac/app/src/main/java/com/youknow/movie/ui/MainActivity.kt
package com.youknow.movie.ui

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.youknow.movie.R
import kotlinx.android.synthetic.main.activity_main.*

const val MOVIE_ID = "movie_id"

class MainActivity : AppCompatActivity() {

    private val TAG = MainActivity::class.java.simpleName

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mainViewPager.adapter = BottomNavigationAdapter(supportFragmentManager)

        bottomNavigation.setOnNavigationItemSelectedListener {
            when (it.itemId) {
                R.id.action_now_playing -> {
                    Log.d(TAG, "[Movies] onNavigationItemSelected - selected: now playing")
                    mainViewPager.currentItem = 0
                    true
                }
                R.id.action_upcoming -> {
                    Log.d(TAG, "[Movies] onNavigationItemSelected - selected: upcoming")
                    mainViewPager.currentItem = 1
                    true
                }
                else -> false
            }
        }

        mainViewPager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

            }

            override fun onPageSelected(position: Int) {
                Log.d(TAG, "[Movies] onPageSelected - $position")
                (bottomNavigation as BottomNavigationView).menu.getItem(position).isChecked = true
            }

            override fun onPageScrollStateChanged(state: Int) {

            }

        })
    }
}